src/test/scala/test_util/BASILTest.scala

```diff
@@ -0,0 +1,136 @@
+                      expectVerify: Boolean,
+                      checkExpected: Boolean = false,
+                      logResults: Boolean = false
+                     )
+
+trait BASILTest extends AnyFunSuite {
+  def runBASIL(inputPath: String, RELFPath: String, specPath: Option[String], BPLPath: String, staticAnalysisConf: Option[StaticAnalysisConfig]): BASILResult = {
+    val specFile = if (specPath.isDefined && File(specPath.get).exists) {
+      specPath
+    } else {
+      None
+    }
+    val config = BASILConfig(
+      loading = ILLoadingConfig(
+        inputFile = inputPath,
+        relfFile = RELFPath,
+        specFile = specFile
+      ),
+      staticAnalysis = staticAnalysisConf,
+      outputPrefix = BPLPath
+    )
+    val result = RunUtils.loadAndTranslate(config)
+    RunUtils.writeOutput(result)
+    result
+  }
+
+  def runBoogie(directoryPath: String, bplPath: String, boogieFlags: Seq[String]): String = {
+    val extraSpec = List.from(File(directoryPath).listFiles()).map(_.toString).filter(_.endsWith(".bpl")).filterNot(_.endsWith(bplPath))
+    val boogieCmd = Seq("boogie", "/printVerifiedProceduresCount:0") ++ boogieFlags ++ Seq(bplPath) ++ extraSpec
+    Logger.info(s"Verifying... ${boogieCmd.mkString(" ")}")
+    val boogieResult = boogieCmd.!!
+    boogieResult
+  }
+
+  /**
+    *
+    * @return param 0: None if passes, Some(failure message) if doesn't pass
+    *         param 1: whether the Boogie output verified
+    *         param 2: whether Boogie timed out
+    */
+  def checkVerify(boogieResult: String, resultPath: String, shouldVerify: Boolean): (Option[String], Boolean, Boolean) = {
+    BASILTest.writeToFile(boogieResult, resultPath)
+    val verified = boogieResult.strip().equals("Boogie program verifier finished with 0 errors")
+    val proveFailed = boogieResult.contains("could not be proved")
+    val timedOut = boogieResult.strip().contains("timed out")
+
+    val failureMsg = if (timedOut) {
+      Some("SMT Solver timed out")
+    } else {
+      (verified, shouldVerify, BASILTest.xor(verified, proveFailed)) match {
+        case (true, true, true) => None
+        case (false, false, true) => None
+        case (_, _, false) => Some("Prover error: unknown result: " + boogieResult)
+        case (true, false, true) => Some("Expected verification failure, but got success.")
+        case (false, true, true) => Some("Expected verification success, but got failure.")
+      }
+    }
+    (failureMsg, verified, timedOut)
+  }
+
+  def checkExpected(expectedOutPath: String, BPLPath: String): (Boolean, Boolean) = {
+    val hasExpected = File(expectedOutPath).exists
+    var matchesExpected = true
+    if (hasExpected) {
+      if (!BASILTest.compareFiles(expectedOutPath, BPLPath)) {
+        matchesExpected = false
+        info("Warning: Boogie file differs from expected")
+      }
+    } else {
+      info("Note: this test has not previously succeeded")
```

Comment:
These info messages are the only thing using functionality from the test suite as far ask I can tell? This could probably be moved back to the SystemTests trait.